integration_test.go

```diff
@@ -141,98 +165,83 @@ var _ = Describe("Integration Testing", func() {
-          "locale": "en",
-          "details": {
-            "app": "or format",
-            "specific": "data..."
-          }
-        }`))
-		Expect(err).To(BeNil())
-	})
+		Describe("disabled HTTP methods", func() {
+			var client = &http.Client{}
+			var url string
 
-	Describe("disabled HTTP methods", func() {
-		var url string
+			BeforeEach(func() {
+				url = testPublishingAPI.URL + "/content/disabled/http/methods"
+			})
 
-		BeforeEach(func() {
-			url = testPublishingAPI.URL + "/content/disabled/http/methods"
-		})
+			It("should not allow GET requets", func() {
+				response, err := client.Get(url)
+				Expect(err).To(BeNil())
+				Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
+			})
 
-		It("should not allow GET requets", func() {
-			response, err := client.Get(url)
-			Expect(err).To(BeNil())
-			Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
-		})
+			It("should not allow POST requets", func() {
+				response, err := client.Post(url, "application/json", bytes.NewBufferString(`{"foo": "bar"}`))
+				Expect(err).To(BeNil())
+				Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
+			})
 
-		It("should not allow POST requets", func() {
-			response, err := client.Post(url, "application/json", bytes.NewBufferString(`{"foo": "bar"}`))
-			Expect(err).To(BeNil())
-			Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
-		})
+			It("should not allow HEAD requets", func() {
+				response, err := client.Head(url)
+				Expect(err).To(BeNil())
+				Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
+			})
 
-		It("should not allow HEAD requets", func() {
-			response, err := client.Head(url)
-			Expect(err).To(BeNil())
-			Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
-		})
+			It("should not allow DELETE requets", func() {
+				request, err := http.NewRequest("DELETE", url, nil)
+				Expect(err).To(BeNil())
 
-		It("should not allow DELETE requets", func() {
-			request, err := http.NewRequest("DELETE", url, nil)
-			Expect(err).To(BeNil())
+				response, err := client.Do(request)
+				Expect(err).To(BeNil())
+				Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
+			})
+		})
 
-			response, err := client.Do(request)
-			Expect(err).To(BeNil())
-			Expect(response.StatusCode).To(Equal(http.StatusMethodNotAllowed))
+		It("returns a 400 error if given invalid JSON", func() {
+			url := testPublishingAPI.URL + "/content" + "/foo/bar"
+			response := DoRequest("PUT", url, []byte("i'm not json"))
+			Expect(response.StatusCode).To(Equal(http.StatusBadRequest))
 		})
 	})
+})
 
-	It("returns a 400 error if given invalid JSON", func() {
-		url := testPublishingAPI.URL + "/content" + "/foo/bar"
-		request, err := http.NewRequest("PUT", url, strings.NewReader("i'm not json"))
-		Expect(err).To(BeNil())
+func DoRequest(verb string, url string, body []byte) (resp *http.Response) {
```

Comment:
Minor style point: It's a little unclear that the return variable is named, but then that name isn't used.  Probably best to not name it in this case:

```go
func DoRequest(verb string, url string, body []byte) *http.Response {
```